ci-operator/step-registry/assisted/baremetal/operator/ztp/assisted-baremetal-operator-ztp-commands.sh

```diff
@@ -39,27 +39,38 @@ source utils.sh
 source network.sh
 
 REPO_DIR="/home/assisted-service"
-if [ ! -d "\${REPO_DIR}" ]; then
-  mkdir -p "\${REPO_DIR}"
+if [ ! -d "${REPO_DIR}" ]; then
+  mkdir -p "${REPO_DIR}"
 
   echo "### Untar assisted-service code..."
-  tar -xzvf /root/assisted-service.tar.gz -C "\${REPO_DIR}"
+  tar -xzvf /root/assisted-service.tar.gz -C "${REPO_DIR}"
 fi
 
-cd "\${REPO_DIR}/deploy/operator/ztp/"
+cd "${REPO_DIR}/deploy/operator/ztp/"
 
 echo "### Deploying spoke cluster..."
 
-export EXTRA_BAREMETALHOSTS_FILE="/root/dev-scripts/\${EXTRA_BAREMETALHOSTS_FILE}"
+export EXTRA_BAREMETALHOSTS_FILE="/root/dev-scripts/${EXTRA_BAREMETALHOSTS_FILE}"
 
 source /root/config
 
-# Inject job configuration for ZTP, if available
+# Inject job configuration for ZTP, if available 
 if [[ -e /root/assisted-ztp-config ]]
 then
   source /root/assisted-ztp-config
 fi
 
-./deploy_spoke_cluster.sh
+# Remove the nodes allocated for Day2 from EXTRA_BAREMETALHOSTS_FILE when deploying the spoke.
+#
+if [ -z "$NUMBER_OF_DAY2_HOSTS" ]
+then
+        cat "${EXTRA_BAREMETALHOSTS_FILE}" > /root/dev-scripts/cluster_bmh.json
+else
+        cat "${EXTRA_BAREMETALHOSTS_FILE}" | jq --arg DAY_2_HOSTS ${NUMBER_OF_DAY2_HOSTS:-0} '.[:-($DAY_2_HOSTS | tonumber)]' > /root/dev-scripts/cluster_bmh.json
```

Comment:
Add a comment above this line that explains that we only use the first (n - NUMBER_OF_DAY2_HOSTS) baremetal hosts for the spoke cluster, and the rest of them are used for day-2 by the "maybe" step

Reply:
Done